python/cudf/cudf/core/frame.py

```diff
@@ -3255,7 +3255,7 @@ def _reindex(
                 # double-argsort to map back from sorted to unsorted positions
                 df = df.take(index.argsort(ascending=True).argsort())
 
-        cols = OrderedDict()
+        cols = {}
```

Comment:
nitpick: feels like this could be made a comprehension instead of initializing then modifying

Reply:
Good call. This comprehension is a bit cumbersome-looking because of the call to `column_empty` in the else clause, but still worth it I think.